src/wallet.cpp

```diff
@@ -1204,7 +1204,7 @@ int64_t CWallet::GetBalance() const
         for (map<uint256, CWalletTx>::const_iterator it = mapWallet.begin(); it != mapWallet.end(); ++it)
         {
             const CWalletTx* pcoin = &(*it).second;
-            if (pcoin->IsTrusted())
+            if (pcoin->IsTrusted() && pcoin->IsConfirmed())
```

Comment:
Balance in the UI is label as "spendable balance", right? If yes then this is ok,

Reply:
Available is what it says. technically u can use the grc after 3 confirmations but everything ui related is 10. we opted for 10 confirms for unconfirmed since the text in wallet said 10.

problem is after 3 confirms without that the available increases and unconfirmed gets added on top till its 10 confirms so for 7 confirms it'll say a higher total balance.

Reply:
illl do some more checks for u thou

Reply:
You can use the grc after 0 confirmations technically.

Reply:
really? i knew if u sent to urself it is trusted immediately. honestly surprised IsTrusted() is not checked before a tx can be used to send coins.

Reply:
I mean, network-wise.
In fact, in Bitcoin it is quite common due to high fees delaying transactions. Impatient receivers can spend the coin immediately, without waiting for confirm and if they set higher fees, both transactions will be confirmed at the same time,

Reply:
ah makes sense